pymc3/plots/energyplot.py

```diff
@@ -36,8 +36,8 @@ def energyplot(trace, kind='kde', figsize=None, ax=None, legend=True,
     try:
         energy = trace['energy']
     except KeyError:
-        print('There is no energy information in the passed trace.')
-        return ax
+        raise ValueError('There is no energy information in the passed trace.')
```

Comment:
you should change the tests, https://github.com/pymc-devs/pymc3/blob/2f2d961e3173cb5a03cc9b2db565f8153c343b8e/pymc3/tests/test_plots.py#L20-L32

Reply:
Oh, funny -- so was `energyplot` just returning empty axes here?

Reply:
Not sure why the code of energy plot had a `print` instead of `raising` and error, but given that was the case, I thought we should test that energy plot handles correctly the absence of the `energy` key... too silly?

Reply:
Hm... this PR shouldn't change the behavior of the code.  I'll swap it back to returning an empty `ax`, but I think a good future PR would be to think about what is proper here.  I'll add a TODO tonight.

Reply:
I guess @fonnesbeck can help with this.

Reply:
For now, went with a warning, and returning the axes.  I think it'd be better to raise an exception, but that can be in the future.  More surprising is that `quantiles` and `hpd` have a decorator that gives them the `squeeze` keyword argument!